File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    let start = 0;
    let end = nums.length - 1;
    let middle;
    while (start < end) {
        middle = Math.floor((start + end) / 2);
        if (nums[middle] < nums[end]) {
            end = middle;
        } else {
            start = middle + 1;
        }
    }
    return nums[start];
};

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.c
int findMin(int* nums, int numsSize) {
    int* start = nums;
    int* end = nums + (numsSize - 1);
    while (start < end) {
        int* middle = start + (end - start) / 2;
        if (*middle < *end) {
            end = middle;
        } else {
            start = middle + 1;
        }
    }
    return *start;
}